websites/A/Animes Portal/presence.ts

```diff
@@ -0,0 +1,365 @@
+      presenceData.largeImageKey = avatar;
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = username;
+    }
+  } else if (pathname.startsWith("/animes")) {
+    const filter: string = new URL(document.location.href).searchParams.get(
+      "filters"
+    );
+    const filters: string[] = [];
+
+    if (filter) {
+      filter
+        .split(",")
+        .map((x: string) => parseInt(x, 10))
+        .forEach((f: number) => {
+          if (filterCodes[f]) filters.push(filterCodes[f]);
+        });
+    }
+
+    if (paths[1] === "search" && paths[2]) {
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")} ${
+        filters.length ? "With " + filters.join(", ") : ""
+      }`;
+    } else if (paths[1]) {
+      presenceData.details = `Viewing animes that starts with the keyboard ${paths[1].toUpperCase()}`;
+    } else
+      presenceData.details = `Viewing animes ${
+        filters.length ? "With " + filters.join(", ") : ""
+      }`;
+  } else if (pathname.startsWith("/anime/")) {
+    const uid = paths[1];
+    const eid = paths[2];
+
+    if (uid && !eid) {
+      const name = document.querySelector(
+        "body > main.animated > div.wrapper > article.rowView > header.rowView-head > h1.heading"
+      ).innerHTML;
+      const image = document
+        .querySelector(
+          "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+        )
+        ?.getAttribute("src");
+
+      if (image) {
+        presenceData.smallImageKey = "logo";
+        presenceData.smallImageText = "logo";
+      }
+
+      presenceData.details = `Viewing ${name}`;
+      presenceData.largeImageKey = image ?? "logo";
+      presenceData.buttons = [
+        {
+          label: `View ${name}`,
+          url: document.location.href
+        }
+      ];
+    } else if (uid && eid) {
+      const { name, episode, part } = getInfo();
+
+      const playing = document
+        .querySelector(
+          "body > main.animated > div.wrapper > section.holder > div.player > div.holder > div.vplayer"
+        )
+        ?.classList.contains("playing");
+
+      if (!playing) {
+        presenceData.details = `Watching ${name}`;
+        presenceData.state = `Episode ${episode}`;
+        presenceData.buttons = [
+          {
+            label: `Watch ${name}`,
+            url: document.location.href
+          }
+        ];
+
+        if (part) presenceData.state = presenceData.state += ` (part ${part})`;
+      } else return;
+    }
+  } else if (pathname.startsWith("/movies")) {
+    const filter: string = new URL(document.location.href).searchParams.get(
+      "filters"
+    );
+    const filters: string[] = [];
+
+    if (filter) {
+      filter
+        .split(",")
+        .map((x: string) => parseInt(x, 10))
+        .forEach((f: number) => {
+          if (filterCodes[f]) filters.push(filterCodes[f]);
+        });
+    }
```

Comment:
Seen this multiple times, why don't you use a function instead of being repetitive?